drivers/adc/adc_stm32f4/src/adc_stm32f4.c

```diff
@@ -0,0 +1,712 @@
+
+    switch (adc_addr) {
+        case (uintptr_t)ADC1:
+        case (uintptr_t)ADC2:
+            switch(cnum) {
+                case ADC_CHANNEL_4:
+                    pin = ADC12_CH4_PIN;
+                    goto done;
+                case ADC_CHANNEL_5:
+                    pin = ADC12_CH5_PIN;
+                    goto done;
+                case ADC_CHANNEL_6:
+                    pin = ADC12_CH6_PIN;
+                    goto done;
+                case ADC_CHANNEL_7:
+                    pin = ADC12_CH7_PIN;
+                    goto done;
+                case ADC_CHANNEL_8:
+                    pin = ADC12_CH8_PIN;
+                    goto done;
+                case ADC_CHANNEL_9:
+                    pin = ADC12_CH9_PIN;
+                    goto done;
+                case ADC_CHANNEL_14:
+                    pin = ADC12_CH14_PIN;
+                    goto done;
+                case ADC_CHANNEL_15:
+                    pin = ADC12_CH15_PIN;
+                    goto done;
+            }
+        case (uintptr_t)ADC3:
+            switch(cnum) {
+                case ADC_CHANNEL_0:
+                    pin = ADC123_CH0_PIN;
+                    goto done;
+                case ADC_CHANNEL_1:
+                    pin = ADC123_CH1_PIN;
+                    goto done;
+                case ADC_CHANNEL_2:
+                    pin = ADC123_CH2_PIN;
+                    goto done;
+                case ADC_CHANNEL_3:
+                    pin = ADC123_CH3_PIN;
+                    goto done;
+                case ADC_CHANNEL_4:
+                    pin = ADC3_CH4_PIN;
+                    goto done;
+                case ADC_CHANNEL_5:
+                    pin = ADC3_CH5_PIN;
+                    goto done;
+                case ADC_CHANNEL_6:
+                    pin = ADC3_CH6_PIN;
+                    goto done;
+                case ADC_CHANNEL_7:
+                    pin = ADC3_CH7_PIN;
+                    goto done;
+                case ADC_CHANNEL_8:
+                    pin = ADC3_CH8_PIN;
+                    goto done;
+                case ADC_CHANNEL_9:
+                    pin = ADC3_CH9_PIN;
+                    goto done;
+                case ADC_CHANNEL_10:
+                    pin = ADC123_CH10_PIN;
+                    goto done;
+                case ADC_CHANNEL_11:
+                    pin = ADC123_CH11_PIN;
+                    goto done;
+                case ADC_CHANNEL_12:
+                    pin = ADC123_CH12_PIN;
+                    goto done;
+                case ADC_CHANNEL_13:
+                    pin = ADC123_CH13_PIN;
+                    goto done;
+                case ADC_CHANNEL_14:
+                    pin = ADC3_CH14_PIN;
+                    goto done;
+                case ADC_CHANNEL_15:
+                    pin = ADC3_CH15_PIN;
+                    goto done;
+            }
+        default:
+            assert(0);
+    }
+done:
+    gpio_td = (GPIO_InitTypeDef){
```

Comment:
I don't believe this is valid C99.  Please do individual structure element assignment to keep this portable.